### ISSUE TITLE
Marker not loading on direct url call

### ISSUE THREAD

**Issue by huntonas** [dburles/meteor-google-maps#37]
I have a site that has one large map with multiple markers. The main page loads fine, but each marker has its own page that has a map on it. When going to the page directly from the marker, the map loads fine and the marker is there. When I go directly to the individual marker page by typing in the url, the map loads fine, but the marker does not appear. I did figure out that if I zoom in/out on the map, the marker will appear. I believe this is because the map refreshes on the zoom and the marker was created after the map was displayed (my best guess). 

Things I have tried:
I tried a trick to refresh the map in the GoogleMaps.read() function that I found on stackoverflow. I do not believe it is hitting the function though when the page is loaded (sometimes). This fix did work once, but every other time, the .ready function was never called or was called before the data was available for the marker. 

I tried map.instance.setZoom(num) in the GoogleMaps.ready() function as well right after creating the marker. This does the same thing as the first one, but figured I would try it. Again, this worked once, but not again. What I am finding is that when the url is accessed directly, the GoogleMaps.ready(0 function is not being called (or the data is not available for the marker), but when I go to it from a link, it works. 

So because of figuring that out, I decided I would try it is different place in the template file. I put all of the code in the onCreated function and in the onRendered function. Neither solved the problem. 

What I believe is going on is that the data is not available for the marker when the function is initially called. When the data is available, the marker is created, but the map does not refresh to show the marker. 

So I tried using Meteor.setTimeout but this didn't solve the issue either. I set it all the way out to 4 seconds, but it still did not work. 

Any ideas? I am happy to send over the url if needed.

**Comment by huntonas** [comment]
I got this working. I have the GoogleMaps.ready() in the onCreated. I have 2 seperate GoogleMaps.ready() functions. The first one adds the marker, and the second does map.instance.setZoom(num). The setZoom refreshed the map, and the marker shows up. Not sure if this is a clean fix or not, but it works for me (and makes the map look animated by the zoom changing in on the marker). I could not get this working by putting the zoom in the same function as the marker. I had to separate them.
